scrapy/http/request/__init__.py

```diff
@@ -50,8 +50,8 @@ def _set_url(self, url):
         if not isinstance(url, six.string_types):
             raise TypeError('Request url must be str or unicode, got %s:' % type(url).__name__)
 
-        url = to_native_str(url, self.encoding)
-        self._url = escape_ajax(safe_url_string(url))
+        s = safe_url_string(url, self.encoding)
```

Comment:
[`FormRequest.from_response` uses response.encoding](https://github.com/scrapy/scrapy/blob/ebef6d7c6dd8922210db8a4a44f48fe27ee0cd16/scrapy/http/request/form.py#L39) already, that's good.